docs/development/setup-vagrant.md

```diff
@@ -147,83 +146,110 @@ The setup for Debian is the same as that [for Ubuntu above](#ubuntu).
 
-```console
-$ git config --global core.symlinks true
-```
+   ```ini
+   NODE_IP_ADDRESS=127.0.0.1
+   NODE_PORT=5672
+   ```
 
-Now confirm the setting:
+1. Run the command below to make sure you are inside the WSL disk and not in a Windows mounted disk.
+   You will run into permission issues if you run `provision` from `zulip`
+   in a Windows mounted disk.
 
-```console
-$ git config core.symlinks
-true
-```
+   ```bash
+   cd ~  # or cd /home/USERNAME
+   ```
 
-If you see `true`, you are ready for [Step 2: Get Zulip
-code](#step-2-get-zulip-code).
+1. [Create your fork](../git/cloning.md#step-1a-create-your-fork) of
+   the [Zulip server repository](https://github.com/zulip/zulip).
 
-Otherwise, if the above command prints `false` or nothing at all, then symlinks
-have not been enabled.
+1. [Create a new SSH key][create-ssh-key] for the WSL-2 Virtual
+   Machine and add it to your GitHub account. Note that SSH keys
+   linked to your Windows computer will not work within the virtual
+   machine.
 
-In **Cygwin**:
+1. Clone and connect to the Zulip upstream repository:
 
-Open a Cygwin window **as an administrator** and do this:
+   ```bash
+   git clone --config pull.rebase git@github.com:YOURUSERNAME/zulip.git ~/zulip
+   cd zulip
+   git remote add -f upstream https://github.com/zulip/zulip.git
+   ```
 
-```console
-christie@win10 ~
-$ echo 'export "CYGWIN=$CYGWIN winsymlinks:native"' >> ~/.bash_profile
-```
+1. Run the following to install the Zulip development environment and
+   start it. (If Windows Firewall creates popups to block services, simply click `Allow Access`.)
 
-Next, close that Cygwin window and open another. If you `echo` $CYGWIN you
-should see:
+   ```bash
+   # Start database, cache, and other services
+   ./tools/wsl/start_services
+   # Install/update the Zulip development environment
+   ./tools/provision
+   # Enter the Zulip Python environment
+   source /srv/zulip-py3-venv/bin/activate
+   # Start the development server
+   ./tools/run-dev.py
+   ```
 
-```console
-christie@win10 ~
-$ echo $CYGWIN
-winsymlinks:native
-```
+   :::{note}
+   If you shut down WSL, after starting it again, you will have to manually start
+   the services using `./tools/wsl/start_services`.
+   :::
 
-Now you are ready for [Step 2: Get Zulip code](#step-2-get-zulip-code).
+1. If you are facing problems or you see error messages after running `./tools/run-dev.py`,
+   you can try running `./tools/provision` again.
+
+1. The [Visual Studio Code Remote -
+   WSL](https://code.visualstudio.com/docs/remote/wsl) extension is
+   recommended for editing files when developing with WSL. When you
+   have it installed, you can run:
+
+   ```bash
+   code .
+   ```
+
+   to open VSCode connected to your WSL environment.
+
+1. You're done! You can pick up the [documentation on using the
+   Zulip development environment](setup-vagrant.md#step-4-developing),
+   ignoring the parts about `vagrant` (since you're not using it).
```

Comment:
Actually, we should probably revise these instructions so that they go through steps 2 and 3. And make subsections if the instructions are distinctly different for Windows users. I think step 2 might be mostly fine and step 3 would need subsections.

Reply:
> I would rewrite this step to be clearer that users here should skip to Step 4.

Yes, I agree.
<img width="721" alt="image" src="https://user-images.githubusercontent.com/2343554/188297269-c3c6c76b-f202-47cb-9c94-bf09a6addee5.png">